lib/operations/execute_operation.js

```diff
@@ -47,45 +45,32 @@ function executeOperation(topology, operation, callback) {
     }
   }
 
-  let result;
-  if (typeof callback !== 'function') {
-    result = new Promise((resolve, reject) => {
-      callback = (err, res) => {
-        if (err) return reject(err);
-        resolve(res);
-      };
-    });
-  }
-
-  function executeCallback(err, result) {
+  function clearSession() {
     if (session && session.owner === owner) {
       session.endSession();
       if (operation.session === session) {
         operation.clearSession();
       }
     }
-
-    callback(err, result);
   }
 
-  try {
-    if (operation.hasAspect(Aspect.EXECUTE_WITH_SELECTION)) {
-      executeWithServerSelection(topology, operation, executeCallback);
-    } else {
-      operation.execute(executeCallback);
+  return maybePromise(callback, cb => {
```

Comment:
refactored `executeOperation` to use `maybePromise`, thought something in here as causing an issue, can revert this.